userspace/falco/app/actions/configure_interesting_sets.cpp

```diff
@@ -105,28 +103,20 @@ static void select_event_set(falco::app::state& s, const libsinsp::events::set<p
 		auto invalid_positive_sc_set_names = unordered_set_difference(user_positive_names, user_positive_sc_set_names);
 		if (!invalid_positive_sc_set_names.empty())
 		{
-			std::cerr << "Invalid (positive) syscall names: warning (base_syscalls override): " + concat_set_in_order(invalid_positive_sc_set_names) << std::endl;
+			falco_logger::log(LOG_WARNING, "Invalid (positive) syscall names: warning (base_syscalls override): "
+				+ concat_set_in_order(invalid_positive_sc_set_names));
 		}
-
-		/* Hidden safety enforcement for `base_syscalls.custom_set` user input override option -> sched_process_exit trace point activation
-		 * (procexit event) is necessary for continuous state engine cleanup, else memory would grow rapidly and linearly over time. */
-		base_sc_set.insert(PPM_SC_SCHED_PROCESS_EXIT);
 	}
 
 	// selected events are the union of the rules events set and the
 	// base events set (either the default or the user-defined one)
 	s.selected_sc_set = rules_sc_set.merge(base_sc_set);
 
-	if (s.config->m_base_syscalls_repair)
-	{
-		/* If base_syscalls.repair set enforce `libsinsp` state based on rules set
-		 * and merge with user supplied base_syscalls.custom_set
-		 *
-		 * Also applies when base_syscalls.custom_set empty but base_syscalls.repair set
-		 * effectively bypassing the default `libsinsp` state enforcement and using
-		 * `sinsp_repair_state_sc_set` instead. */
-		s.selected_sc_set = libsinsp::events::sinsp_repair_state_sc_set(rules_sc_set).merge(base_sc_set);
-	}
+	/* Hidden safety enforcement for `base_syscalls.custom_set` user input
+	 * override option -> sched_process_exit trace point activation
+	 * (procexit event) is necessary for continuous state engine cleanup,
+	 * else memory would grow rapidly and linearly over time. */
+	s.selected_sc_set.insert(ppm_sc_code::PPM_SC_SCHED_PROCESS_EXIT);
```

Comment:
We need to repair procexit regardless on whether we use a custom set or not, just as a safety measure at this point.

Reply:
Agree we should really do this safety just in case always, it's one insert only ...